eth/peerset.go

```diff
@@ -148,6 +159,40 @@ func (ps *peerSet) registerDiffExtension(peer *diff.Peer) error {
 	return nil
 }
 
+// registerTrustExtension unblocks an already connected `eth` peer waiting for its
+// `trust` extension, or if no such peer exists, tracks the extension for the time
+// being until the `eth` main protocol starts looking for it.
+func (ps *peerSet) registerTrustExtension(peer *trust.Peer) error {
+	// Reject the peer if it advertises `trust` without `eth` as `trust` is only a
+	// satellite protocol meaningful with the chain selection of `eth`
+	if !peer.RunningCap(eth.ProtocolName, eth.ProtocolVersions) {
+		return errTrustWithoutEth
+	}
+	// If the peer isn't verify node, don't register trust extension into eth protocol.
+	if !peer.VerifyNode() {
+		return nil
```

Comment:
Do we need to:
```
delete(ps.trustWait, id)
wait <- peer
```

So that the eth protocol won't wait any more.

Reply:
Look the waitTrustExtension function, if the peer isn't a verify node, eth protocol won't wait, so I think it is unnecessary to do what you mentioned here.

Reply:
cool